hcl2template/types.packer_config.go

```diff
@@ -320,41 +320,53 @@ func (cfg *PackerConfig) getCoreBuildProvisioners(source SourceUseBlock, blocks
 	var diags hcl.Diagnostics
 	res := []packer.CoreBuildProvisioner{}
 	for _, pb := range blocks {
-		if pb.OnlyExcept.Skip(source.String()) {
-			continue
-		}
-		provisioner, moreDiags := cfg.startProvisioner(source, pb, ectx)
+		skip, coreBuildProv, moreDiags := cfg.getCoreBuildProvisioner(source, pb, ectx)
 		diags = append(diags, moreDiags...)
-		if moreDiags.HasErrors() {
+		if moreDiags.HasErrors() || skip {
 			continue
 		}
+		res = append(res, coreBuildProv)
+	}
+	return res, diags
+}
 
-		// If we're pausing, we wrap the provisioner in a special pauser.
-		if pb.PauseBefore != 0 {
-			provisioner = &packer.PausedProvisioner{
-				PauseBefore: pb.PauseBefore,
-				Provisioner: provisioner,
-			}
-		} else if pb.Timeout != 0 {
-			provisioner = &packer.TimeoutProvisioner{
-				Timeout:     pb.Timeout,
-				Provisioner: provisioner,
-			}
+func (cfg *PackerConfig) getCoreBuildProvisioner(source SourceUseBlock, pb *ProvisionerBlock, ectx *hcl.EvalContext) (skip bool, coreBuildProv packer.CoreBuildProvisioner, diags hcl.Diagnostics) {
+	if pb.OnlyExcept.Skip(source.String()) {
+		skip = true
+		return
+	}
```

Comment:
Here I think that the skip should be higher: 
```suggestion
func (cfg *PackerConfig) getCoreBuildProvisioner(source SourceUseBlock, pb *ProvisionerBlock, ectx *hcl.EvalContext) (coreBuildProv packer.CoreBuildProvisioner, diags hcl.Diagnostics) {
```

Reply:
( for context: I think in terms of naming and usage, getCoreBuildProvisioner should try to get you a provisioner or 'fail' and that the control for the calling loop should be higher )

Reply:
I agree!! I had this question "in or out" and end up with the first but I think you're right!